server/jms/src/main/java/org/hawkular/apm/server/jms/RetryCapableMDB.java

```diff
@@ -24,13 +24,18 @@
 import javax.jms.MessageListener;
 import javax.jms.TextMessage;
 
+import org.hawkular.apm.api.services.Publisher;
+import org.hawkular.apm.server.api.task.Handler;
+import org.hawkular.apm.server.api.task.ProcessingUnit;
+import org.hawkular.apm.server.api.task.Processor;
+
 import com.fasterxml.jackson.core.type.TypeReference;
 import com.fasterxml.jackson.databind.ObjectMapper;
 
 /**
  * @author gbrown
  */
-public abstract class RetryCapableMDB<S> implements MessageListener {
+public abstract class RetryCapableMDB<S,T> implements MessageListener {
```

Comment:
Please, add a javadoc explaining the parameters. I really don't know what Sun engineers had in mind when they decided that single letter type parameter names were a good idea... :)